src/transformers/models/vits/modeling_vits.py

```diff
@@ -0,0 +1,1491 @@
+# coding=utf-8
+# Copyright 2023 The VITS Authors and the HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch VITS model."""
+
+import math
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.utils.checkpoint
+from torch import nn
+
+from ...activations import ACT2FN
+from ...deepspeed import is_deepspeed_zero3_enabled
+from ...modeling_outputs import (
+    BaseModelOutput,
+    ModelOutput,
+)
+from ...modeling_utils import PreTrainedModel
+from ...utils import add_start_docstrings, add_start_docstrings_to_model_forward, logging, replace_return_docstrings
+from .configuration_vits import VitsConfig
+
+
+logger = logging.get_logger(__name__)
+
+
+# General docstring
+_CONFIG_FOR_DOC = "VitsConfig"
+
+
+VITS_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "Matthijs/mms-tts-eng",
+    # See all VITS models at https://huggingface.co/models?filter=vits
+]
+
+
+@dataclass
+class VitsModelOutput(ModelOutput):
+    """
+    Describes the outputs for the VITS model, with potential hidden states and attentions.
+
+    Args:
+        audio (`torch.FloatTensor` of shape `(batch_size, sequence_length)`):
```

Comment:
WDYT about `output_values`? We use `input_values` to describe an input waveform to our CTC models, so `output_values` is consistent here?

Reply:
`output_values` is a bit ambiguous - everything returned is an output value. What do you think about just `last_hidden_state` if we don't want to specify the audio part?

Reply:
Resolved in https://github.com/huggingface/transformers/pull/24085/commits/2cfd761230983d335ad6c4de1dafc52bff3c8000

Reply:
Okay `waveform` better in this case! `last_hidden_state` isn't strictly correct since we're not returning a latent, but rather a real-valued temporal array